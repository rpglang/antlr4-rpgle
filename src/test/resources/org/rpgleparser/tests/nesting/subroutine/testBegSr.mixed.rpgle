       DCL-S X Int(10);
       EXSR SUBRT1;
       BEGSR SUBRT1;                                      
         X = 1234;
     C                   ENDSR     SUBRT1                                      