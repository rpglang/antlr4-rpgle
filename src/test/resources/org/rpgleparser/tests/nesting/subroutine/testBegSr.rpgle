       DCL-S X Int(10);
       BEGSR SomeSubRoutine;
         X = 1234;
       ENDSR;