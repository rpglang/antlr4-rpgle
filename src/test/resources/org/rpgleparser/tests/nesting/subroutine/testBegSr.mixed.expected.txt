/*===TOKENS===*/
DS_Standalone            <DCL-S>
ID                       <X>
INT                      <Int>
'('                      <(>
NUMBER                   <10>
')'                      <)>
FREE_SEMI                <;>
OP_EXSR                  <EXSR>
ID                       <SUBRT1>
FREE_SEMI                <;>
OP_BEGSR                 <BEGSR>
ID                       <SUBRT1>
FREE_SEMI                <;>
ID                       <X>
'='                      <=>
NUMBER                   <1234>
FREE_SEMI                <;>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_ENDSR                 <ENDSR>
CS_FactorContent         <SUBRT1>
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
/*===TREE===*/
(r
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword X)))
      (datatype (datatypeName Int) (args ( (expression (number 10)) )))
      ;
    )
  )
  (statement
    (free
      (baseExpression
        (op (op_exsr EXSR (identifier (free_identifier (idOrKeyword SUBRT1)))))
      )
      ;
    )
  )
  (subroutine
    (begsr (freeBEGSR BEGSR (identifier (free_identifier (idOrKeyword SUBRT1))) ;))
    (statement
      (free
        (baseExpression
          (op
            (op_eval
              (evalExpression
                (assignmentExpression
                  (simpleExpression (identifier (free_identifier (idOrKeyword X))))
                  =
                  (expression (number 1234))
                )
              )
            )
          )
        )
        ;
      )
    )
    (endsr
      (csENDSR
        C
        (cs_controlLevel '  ')
        (onOffIndicatorsFlag ' ')
        (cs_indicators '  ')
        (factor '              ')
        ENDSR
        (cspec_fixed_standard_parts
          (factor (factorContent SUBRT1))
          (resultType '              ')
          '     '
          '  '
          (resultIndicator '  ')
          (resultIndicator '  ')
          (resultIndicator '  ')
          <EOF>
        )
      )
    )
  )
)
/*======*/